service.c

```diff
@@ -496,6 +496,7 @@ int release_services()
                 ci_list_destroy(service_extra_data_list[i].option_handlers);
                 service_extra_data_list[i].option_handlers = NULL;
             }
+            free(service_data(service_list[i]));
```

Comment:
This is does not looks correct. The service_data function will return a pointer to an item of the service_extra_data_list array. This item is not an allocated memory block.

In the other hand the service_extra_data_list array needs to be released.